# Comprensión de los Datos

In [2]:
#importa librerías
import pandas as pd

# Descripción de Variables


Pclass Passenger Class (1 = 1st; 2 = 2nd; 3 = 3rd): Categórica Nominal 
survival Survival (0 = No; 1 = Yes)  
name Name  
sex Sex  
age Age  
sibsp Number of Siblings/Spouses Aboard  
parch Number of Parents/Children Aboard  
ticket Ticket Number  
fare Passenger Fare (British pound)  
cabin Cabin  
embarked Port of Embarkation (C = Cherbourg; Q = Queenstown; S = Southampton)  
boat Lifeboat  
body Body Identification Number  
home.dest Home/Destination

**Ejemplo:** Crear un objeto DataFrame con base en un archivo .csv (poner "df = ")

In [3]:
#lee archivo csv
df = pd.read_csv("diabetes.csv")

In [5]:
#Usa función shape para revisar el total de renglones y columnas
df.shape

(768, 9)

In [6]:
#Revisa los primeros 5 renglones del dataset usando la función head()
df.head()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1

In [7]:
#Revisa los últimos 5 renglones del dataset usando la función tail()
df.tail()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
763           10      101             76             48      180  32.9   
764            2      122             70             27        0  36.8   
765            5      121             72             23      112  26.2   
766            1      126             60              0        0  30.1   
767            1       93             70             31        0  30.4   

     DiabetesPedigreeFunction  Age  Outcome  
763                     0.171   63        0  
764                     0.340   27        0  
765                     0.245   30        0  
766                     0.349   47        1  
767                     0.315   23        0

In [8]:
#Revisa la información mas completa del conjunto de datos usando la función info()
#Muestra el total de datos, las columnas y su tipo correspondiente, dice si contiene nulos o no
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 768 entries, 0 to 767
Data columns (total 9 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Pregnancies               768 non-null    int64  
 1   Glucose                   768 non-null    int64  
 2   BloodPressure             768 non-null    int64  
 3   SkinThickness             768 non-null    int64  
 4   Insulin                   768 non-null    int64  
 5   BMI                       768 non-null    float64
 6   DiabetesPedigreeFunction  768 non-null    float64
 7   Age                       768 non-null    int64  
 8   Outcome                   768 non-null    int64  
dtypes: float64(2), int64(7)
memory usage: 54.1 KB


In [9]:
#revisa cuántos valores únicos tiene cada atributo del archivo usando la función nunique()
df.nunique()

Pregnancies                  17
Glucose                     136
BloodPressure                47
SkinThickness                51
Insulin                     186
BMI                         248
DiabetesPedigreeFunction    517
Age                          52
Outcome                       2
dtype: int64

## Exploración de Datos

In [10]:
#utiliza la función describe() para obtener estadística básica. se puede incluir -0
df.describe()

Pregnancies     Glucose  BloodPressure  SkinThickness     Insulin  \
count   768.000000  768.000000     768.000000     768.000000  768.000000   
mean      3.845052  120.894531      69.105469      20.536458   79.799479   
std       3.369578   31.972618      19.355807      15.952218  115.244002   
min       0.000000    0.000000       0.000000       0.000000    0.000000   
25%       1.000000   99.000000      62.000000       0.000000    0.000000   
50%       3.000000  117.000000      72.000000      23.000000   30.500000   
75%       6.000000  140.250000      80.000000      32.000000  127.250000   
max      17.000000  199.000000     122.000000      99.000000  846.000000   

              BMI  DiabetesPedigreeFunction         Age     Outcome  
count  768.000000                768.000000  768.000000  768.000000  
mean    31.992578                  0.471876   33.240885    0.348958  
std      7.884160                  0.331329   11.760232    0.476951  
min      0.000000                  0.078000   21.000000    0.000000  
25%     27.300000                  0.243750   24.000000    0.000000  
50%     32.000000                  0.372500   29.000000    0.000000  
75%     36.600000                  0.626250   41.000000    1.000000  
max     67.100000                  2.420000   81.000000    1.000000

In [14]:
#Revisa Valores nulos con funcion isnull().sum()
df.isnull().sum()

Pregnancies                 0
Glucose                     0
BloodPressure               0
SkinThickness               0
Insulin                     0
BMI                         0
DiabetesPedigreeFunction    0
Age                         0
Outcome                     0
dtype: int64

In [16]:
#Revisar valores únicos por columna usando función unique(): nombre-columna.unique()
df.Pregnancies.unique()

array([ 6,  1,  8,  0,  5,  3, 10,  2,  4,  7,  9, 11, 13, 15, 17, 12, 14])

In [17]:
#Revisar valores únicos por columna usando función unique(): nombre-columna.unique()
df.Glucose.unique()

array([148,  85, 183,  89, 137, 116,  78, 115, 197, 125, 110, 168, 139,
       189, 166, 100, 118, 107, 103, 126,  99, 196, 119, 143, 147,  97,
       145, 117, 109, 158,  88,  92, 122, 138, 102,  90, 111, 180, 133,
       106, 171, 159, 146,  71, 105, 101, 176, 150,  73, 187,  84,  44,
       141, 114,  95, 129,  79,   0,  62, 131, 112, 113,  74,  83, 136,
        80, 123,  81, 134, 142, 144,  93, 163, 151,  96, 155,  76, 160,
       124, 162, 132, 120, 173, 170, 128, 108, 154,  57, 156, 153, 188,
       152, 104,  87,  75, 179, 130, 194, 181, 135, 184, 140, 177, 164,
        91, 165,  86, 193, 191, 161, 167,  77, 182, 157, 178,  61,  98,
       127,  82,  72, 172,  94, 175, 195,  68, 186, 198, 121,  67, 174,
       199,  56, 169, 149,  65, 190])

In [18]:
#Revisar valores únicos por columna usando función unique(): nombre-columna.unique()
df.Outcome.unique()

array([1, 0])

## Variables Cuantitativas

### Medidas de tendencia central


In [20]:
#Pregnancies
#Se puede obtener la media, mediana y moda para
mean_pregnancies = df['Pregnancies'].mean()
median_pregnancies = df['Pregnancies'].median()
mode_pregnancies = df['Pregnancies'].mode()
print("Mean_pregnancies:",mean_pregnancies)
print("Median_pregnancies:",median_pregnancies)
print("Mode_pregnancies:",mode_pregnancies)

Mean_pregnancies: 3.8450520833333335
Median_pregnancies: 3.0
Mode_pregnancies: 0    1
Name: Pregnancies, dtype: int64


Conclusiones:
El promedio de embarazos fue de 3  
La cantidad de embarazos al centro es 3  
La cantidad de embarazos más repetida fue de 0 y 1

In [22]:
#Glucose
#Se puede obtener la media, mediana y moda para
mean_glucose = df['Glucose'].mean()
median_glucose = df['Glucose'].median()
mode_glucose = df['Glucose'].mode()
print("Mean_glucose:",mean_glucose)
print("Median_glucose:",median_glucose)
print("Mode_glucose:",mode_glucose)

Mean_glucose: 120.89453125
Median_glucose: 117.0
Mode_glucose: 0     99
1    100
Name: Glucose, dtype: int64


Conclusiones:  
El promedio de glucosa fue de 120  
La glucosa al centro es 117  
La glucosa más repetida fue de 0, 1, 99 y 100

In [24]:
#Outcome
#Se puede obtener la media, mediana y moda para
mean_outcome = df['Outcome'].mean()
median_outcome = df['Outcome'].median()
mode_outcome = df['Outcome'].mode()
print("Mean_outcome:",mean_outcome)
print("Median_outcome:",median_outcome)
print("Mode_outcome:",mode_outcome)

Mean_outcome: 0.3489583333333333
Median_outcome: 0.0
Mode_outcome: 0    0
Name: Outcome, dtype: int64


Conclusiones:
El promedio de resultado fue de 0.3  
El resultado al centro es 0  
El resultado más repetido fue de 0

Cabe mencionar que, en las discusiones en Kaggle, se menciona que el dato "0" significa que el paciente es sano y el dato "1" significa que el paciente tiene diabetes. Por lo tanto, la mayoría de los datos indican un paciente sano.

# Variables Categóricas

In [21]:
#Para conteo  de cada valor en una columna, en orden descendente usar función value_counts():    
# nombreDataframe.columna.value_counts()
# nombreDataframe['columna'].value_counts()
df.Pregnancies.value_counts()

Pregnancies
1     135
0     111
2     103
3      75
4      68
5      57
6      50
7      45
8      38
9      28
10     24
11     11
13     10
12      9
14      2
15      1
17      1
Name: count, dtype: int64

In [25]:
#Para conteo  de cada valor en una columna, en orden descendente usar función value_counts():    
# nombreDataframe.columna.value_counts()
# nombreDataframe['columna'].value_counts()
df.Glucose.value_counts()

Glucose
99     17
100    17
111    14
129    14
125    14
       ..
191     1
177     1
44      1
62      1
190     1
Name: count, Length: 136, dtype: int64

In [26]:
#Para conteo  de cada valor en una columna, en orden descendente usar función value_counts():    
# nombreDataframe.columna.value_counts()
# nombreDataframe['columna'].value_counts()
df.Outcome.value_counts()

Outcome
0    500
1    268
Name: count, dtype: int64

In [27]:
#Para conteo  de cada valor en una columna, en orden descendente usar función value_counts():    
# nombreDataframe.columna.value_counts()
# nombreDataframe['columna'].value_counts()
df["Pregnancies"].value_counts()

Pregnancies
1     135
0     111
2     103
3      75
4      68
5      57
6      50
7      45
8      38
9      28
10     24
11     11
13     10
12      9
14      2
15      1
17      1
Name: count, dtype: int64

In [28]:
#Para conteo  de cada valor en una columna, en orden descendente usar función value_counts():    
# nombreDataframe.columna.value_counts()
# nombreDataframe['columna'].value_counts()
df["Glucose"].value_counts()

Glucose
99     17
100    17
111    14
129    14
125    14
       ..
191     1
177     1
44      1
62      1
190     1
Name: count, Length: 136, dtype: int64

In [29]:
#Para conteo  de cada valor en una columna, en orden descendente usar función value_counts():    
# nombreDataframe.columna.value_counts()
# nombreDataframe['columna'].value_counts()
df["Outcome"].value_counts()

Outcome
0    500
1    268
Name: count, dtype: int64

In [10]:
#Revisa conteo de varias columnas

In [42]:
# Crear variable totalPregDiabetic que incluya la suma de las columnas Pregnancies y Outcome con valor "1"
# Mostrar el total por cada tamaño de familia
conteo_preg = df["Pregnancies"].count()
conteo_outcome = (df["Outcome"] == 1).sum()
df["totalPregDiabetic"] = df["Pregnancies"] + (df["Outcome"] == 1).astype(int)

In [43]:
df.head

<bound method NDFrame.head of      Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0              6      148             72             35        0  33.6   
1              1       85             66             29        0  26.6   
2              8      183             64              0        0  23.3   
3              1       89             66             23       94  28.1   
4              0      137             40             35      168  43.1   
..           ...      ...            ...            ...      ...   ...   
763           10      101             76             48      180  32.9   
764            2      122             70             27        0  36.8   
765            5      121             72             23      112  26.2   
766            1      126             60              0        0  30.1   
767            1       93             70             31        0  30.4   

     DiabetesPedigreeFunction  Age  Outcome  totalPregDiabetic  
0               

## Consulta

In [44]:
# df.iloc[i]: Accede a la fila en la posición i.
# Acceder a la primera fila
df.iloc[0]

Pregnancies                   6.000
Glucose                     148.000
BloodPressure                72.000
SkinThickness                35.000
Insulin                       0.000
BMI                          33.600
DiabetesPedigreeFunction      0.627
Age                          50.000
Outcome                       1.000
totalPregDiabetic             7.000
Name: 0, dtype: float64

In [45]:
# Acceder a las dos primeras filas
df.iloc[:3]

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   

   DiabetesPedigreeFunction  Age  Outcome  totalPregDiabetic  
0                     0.627   50        1                  7  
1                     0.351   31        0                  1  
2                     0.672   32        1                  9

In [46]:
#Seleccionar columnas, indicando entre corchetes [nombreColumna, nombreColumna]
df[["Pregnancies", "Glucose"]]

Pregnancies  Glucose
0              6      148
1              1       85
2              8      183
3              1       89
4              0      137
..           ...      ...
763           10      101
764            2      122
765            5      121
766            1      126
767            1       93

[768 rows x 2 columns]

In [50]:
#Selección de filas [indicar dataframe[columna] operador valor]
embarazos = df[df["Pregnancies"] == 0]

In [51]:
#ordenar usando funcion sort_values(by=atributo, ascending=True/false
embarazos.sort_values(by="Pregnancies", ascending=True)

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
4              0      137             40             35      168  43.1   
528            0      117             66             31      188  30.8   
511            0      139             62             17      210  22.1   
506            0      180             90             26       90  36.5   
487            0      173             78             32      265  46.5   
..           ...      ...            ...            ...      ...   ...   
247            0      165             90             33      680  52.3   
239            0      104             76              0        0  18.4   
237            0      179             90             27        0  44.1   
297            0      126             84             29      215  30.7   
757            0      123             72              0        0  36.3   

     DiabetesPedigreeFunction  Age  Outcome  totalPregDiabetic  
4                       2.288   33        1                  1  
528                     0.493   22        0                  0  
511                     0.207   21        0                  0  
506                     0.314   35        1                  1  
487                     1.159   58        0                  0  
..                        ...  ...      ...                ...  
247                     0.427   23        0                  0  
239                     0.582   27        0                  0  
237                     0.686   23        1                  1  
297                     0.520   24        0                  0  
757                     0.258   52        1                  1  

[111 rows x 10 columns]

In [52]:
#Agrupar por un atributo y calcular función de agregación utilizando groupby(atributo)['atributoAgregacion].funcion() (ej. mean() para promedio)
embarazos.groupby("Glucose")["Outcome"].mean()

Glucose
57     0.0
67     0.0
73     0.0
74     0.0
78     0.0
      ... 
180    1.0
181    1.0
188    1.0
189    1.0
198    1.0
Name: Outcome, Length: 64, dtype: float64

Crea un subconjunto de **glucose** para el valor mayor a 180

In [56]:
# usa el criterio para extraer solo los boletos caros con fare > 50
embarazos[embarazos["Glucose"] > 180]

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
440            0      189            104             25        0  34.3   
561            0      198             66             32      274  41.3   
595            0      188             82             14      185  32.0   
753            0      181             88             44      510  43.3   

     DiabetesPedigreeFunction  Age  Outcome  totalPregDiabetic  
440                     0.435   41        1                  1  
561                     0.502   28        1                  1  
595                     0.682   22        1                  1  
753                     0.222   26        1                  1